spec/models/document_spec.rb

```diff
@@ -833,4 +833,34 @@ def initialize(params = {})
       expect(AaibReport.title).to eq("AAIB Report")
     end
   end
+
+  describe "#send_email_on_publish?" do
+    it "should send email if document is major update and no disable_email_alert override exists" do
+      doc = MyDocumentType.new(update_type: "major")
+      expect(doc.send_email_on_publish?).to be true
+    end
+
+    it "should send email if document is major update and disable_email_alert override is false" do
+      doc = MyDocumentType.new(update_type: "major")
+      doc.disable_email_alert = false
+      expect(doc.send_email_on_publish?).to be true
+    end
+
+    it "(Caution) will stop sending email if document is major update and disable_email_alert override is a random string" do
+      doc = MyDocumentType.new(update_type: "major")
+      doc.disable_email_alert = "invalid"
+      expect(doc.send_email_on_publish?).to be false
+    end
```

Comment:
Hmmm could we have `send_email_on_publish?` raise an exception, rather than semi-support using strings? Better for it to crash than to silently do something we're not expecting, IMO.

Reply:
i was on the fence about this - thought it wasn't likely to happen since the rake task passes bools anyways, so should be enough to call out as a test - but yes it would be safer. Will add that in

Reply:
Tried implementing an exception raising and it feels like it doesn't read so well. 
```
  def send_email_on_publish?
    raise "Invalid disable email alert flag. Please use booleans only." if disable_email_alert.present && ![true, false].include?(disable_email_alert)
    update_type == "major" && !disable_email_alert
  end
```

I'm not sure this is necessary as the usage is internal code and not exposed to any external API or users. I feel like as long as we document this behaviour in the test, it should be enough. Will leave that discussion for next time and amend as a separate PR if needed.

Reply:
Discussion happened - putting it in